File: codeforces/round379/A.cc
// Codeforces Round #379 (Div. 2)
// http://codeforces.com/contest/734

// ID: 734A (Anton and Danik)

#include <iostream>
#include <stdio.h>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;

void solve() {
	int n;
	string s;
	cin >> n >> s;

	int a = 0, d = 0;
	for(int i=0; i<n; i++){
		if(s[i] == 'A') a++;
		else d++;
	}

	if(a > d) cout << "Anton" << endl;
	else if(a < d) cout << "Danik" << endl;
	else cout << "Friendship" << endl;
}

int main() {
	solve();
	return 0;
}

File: codeforces/round379/A.hs
-- Codeforces Round #379 (Div. 2)
-- http://codeforces.com/contest/734

-- ID: 734A (Anton and Danik)

main = do
	n <- read `fmap` getLine :: IO Int
	s <- getLine
	let anton = length $ filter (== 'A') s
	let danik = length $ filter (== 'D') s
	if anton == danik
	then putStrLn "Friendship"
	else if anton > danik
	then putStrLn "Anton"
	else putStrLn "Danik"
